packages/input-option/src/InputOptionContent/InputOptionContent.tsx

```diff
@@ -1,54 +1,93 @@
 import React from 'react';
 
 import { cx } from '@leafygreen-ui/emotion';
-import { Description } from '@leafygreen-ui/typography';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { Description, Label } from '@leafygreen-ui/typography';
 
 import {
   descriptionClassName,
-  titleClassName,
-} from '../InputOption/InputOption.style';
-import {
-  contentWrapper,
-  descriptionBaseStyles,
-  glyphContainer,
-  glyphRightStyles,
+  getContentWrapperStyles,
+  getDescriptionStyles,
+  getLeftGlyphStyles,
+  getRightGlyphStyles,
+  getTitleStyles,
+  inputOptionContentClassName,
   leftGlyphClassName,
-  textWrapper,
-  titleBaseStyles,
+  textContainerStyles,
+  titleClassName,
 } from '../InputOptionContent/InputOptionContent.styles';
+import { useInputOptionContext } from '../InputOptionContext';
 
 import { InputOptionContentProps } from './InputOptionContent.types';
 
 /**
  * @internal
  *
- * This is a temp workaround to add consistent option styles. Once all components that use an input option are consistent we can add this directly inside `InputOption`.
+ * This is a temp workaround to add consistent option styles.
+ * Once all components that use an input option are consistent
+ * we can add this directly inside `InputOption`.
  */
 export const InputOptionContent = ({
   children,
   description,
   leftGlyph,
   rightGlyph,
+  preserveIconSpace = true,
+  className,
+  ...rest
 }: InputOptionContentProps) => {
+  const { disabled, highlighted, darkMode } = useInputOptionContext();
+  const { theme } = useDarkMode(darkMode);
   return (
-    <div className={contentWrapper}>
+    <div
+      className={cx(
+        inputOptionContentClassName,
+        getContentWrapperStyles({
+          hasLeftGlyph: !!leftGlyph || preserveIconSpace,
+          hasRightGlyph: !!rightGlyph,
+        }),
+        className,
+      )}
+      {...rest}
+    >
       {leftGlyph && (
-        <div className={cx(leftGlyphClassName, glyphContainer)}>
+        <div
+          className={cx(
+            leftGlyphClassName,
+            getLeftGlyphStyles({ theme, disabled, highlighted }),
+          )}
+        >
           {leftGlyph}
         </div>
       )}
-      <div className={textWrapper}>
-        <div className={cx(titleClassName, titleBaseStyles)}>{children}</div>
+      <div className={textContainerStyles}>
+        <Label
+          as="div"
+          htmlFor={''}
```

Comment:
is this correct?

Reply:
maybe we should just not use the label component here?

Reply:
I guess it is a label IF it's an input option, but if we're using it _not_ as one then this doesn't make sense

Reply:
Yeah I was using this to ensure we got styles consistent with the `Label` component. Might be more correct to just re-write them here